boot-core/src/daemon/core.rs

```diff
@@ -250,6 +253,21 @@ impl TxHandler for Daemon {
     }
 }
 
+impl<T: BootExecute<Daemon> + ContractInstance<Daemon> + Clone> CallAs<Daemon> for T {
+    type Sender = Wallet;
+
+    fn set_sender(&mut self, sender: &Self::Sender) -> &mut Self {
```

Comment:
Would prefer `with_sender` implying that it allows for chaining but lgtm!

Reply:
can update it to remove the `&mut Self` return, will have a look in our codebase if we actually use the return ever.